### ISSUE TITLE
Articles categorize implementation

### ISSUE THREAD

**Issue by isuriupeksha** [systers/language-translation#33]
Currently user has to add category name (`string`) value every time the category when user create a article. So I am going to implement the category management feature. so that user can select existing category or add new category when create new article.

It associated by  `article has_one category`. 

-------------

Concerns:
1. Categories can only created, update and delete by `Admins`
2. Others can add into created article.

Any suggestion?

**Comment by wonook** [comment]
I think it would be more convenient to have autocompletion for the categories. Some of the default/example categories can be added on the `seeds.rb` file.

**Comment by isuriupeksha** [comment]
@wonook  Ok, then what are the default categories? I would like to add this JS library to replace default browser select menu. https://select2.github.io/

**Comment by wonook** [comment]
Why not migrate our front-end to `React` instead of adding JS libraries? I think it'll be better the earlier. We can implement your designs there without having to do the work twice. #11 
I think we'll be able to do a LOT more things with the front-end of the application by doing that.

**Comment by isuriupeksha** [comment]
Great! We can discuss later when we develop front-end application. It is not required to add more JS libraries. I know that adding larger JS library for small feature doesn't good. But I like that plugin :)

**Comment by wonook** [comment]
Yeah, but with my experience with AngularJS, there are a LOT more libraries out there that makes better use of JS (for autocompletion, etc.). I'll try to do #11 ASAP upon finishing up #30. Then, we'll be able to take advantage of the powerful JS features. Let's put this issue on hold until then. Does that sound okay for you?

**Comment by isuriupeksha** [comment]
yes. :)

**Comment by wonook** [comment]
Alright, I'll get my mind on that right after #41 has been merged.

**Comment by isuriupeksha** [comment]
@wonook we need to add more test cases by user role.
